components/learn/modules/ROOT/pages/deploy-and-interact.adoc

```diff
@@ -186,17 +186,147 @@ TIP: To learn more about `send-tx` and `call`, check out the https://docs.openze
 [[interacting-programatically]]
 == Interacting Programatically
 
-from sdk interacting guide (https://docs.openzeppelin.com/sdk/2.6/interacting#setup)
+While interacting via the command line is useful for running quick tests, in the end you will want to interact with your contracts from your own code. Here, we will see how to use https://web3js.readthedocs.io/en/1.2.4/[web3.js] to do so, so let's install it.
 
-replace manual setup with contract loader?
+[source,console]
+----
+npm install web3
+----
 
-send tx, call function, same as CLI guide
+TIP: Keep in mind that there are many other javascript libraries available, and you can use whichever you like the most. Once a contract is deployed, you can interact with it through any library!
 
+[[connecting-to-the-network]]
+=== Connecting to the Network
 
-== closing thoughts ???
+Let's begin coding in a new `src/index.js` file, where we will be writing our javascript script. We will start with some boilerplate for writing async code, and setting up a new `web3` object.
 
-this was was like testing, but manual and not very complete. automated is better.
+[source,js]
+----
+const Web3 = require('web3');
 
-next steps:
-    - automated testing
-    - deploying on public testnets
+async function main() {
+  // Set up web3 object, connected to the local development network
+  const web3 = new Web3('http://localhost:8545');
+}
+
+main();
+----
+
+Here, we are initializing a new `web3` instance connecting to the local development network we started earlier, that is running on localhost port 8545. We can test if the connection works by asking something to the local node, such as the list of enabled accounts:
+
+[source,js]
+----
+// Set up web3 object, connected to the local development network
+const web3 = new Web3('http://localhost:8545');
+
+// Retrieve accounts from the local node
+const accounts = await web3.eth.getAccounts();
+console.log(accounts);
+----
+
+NOTE: We will not be repeating the boilerplate code on every snippet, but make sure to always code _inside_ the `main` function we defined above!
+
+Run the code above using `node`, and check that you are getting a list of available accounts in response. 
+
+[source,console]
+----
+$ node src/index.js 
+[ '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1',
+  '0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0',
+  ... ]
+----
+
+These accounts should match the ones you got when you ran `openzeppelin accounts` earlier. Now that we have our first code snippet for getting data out of a blockchain, let's start working with our contract.
+
+[[getting-a-contract-instance]]
+=== Getting a contract instance
```

Comment:
```suggestion
=== Getting a Contract Instance
```

Apparently title capitalization is rather important. I use https://capitalizemytitle.com/ when I'm unsure what the proper capitalization should be.